File: lab/3-4/lab_3-4.py
def one():
    n = int(input())
    print(f'Число\tКвадрат')
    print('-'*15)
    
    for i in range(1, n+1, 2):
        print(f'{i}\t{i**2}')


def two():
    a, b, c = [int(input()) for i in range(3)]
    counter = 0
    
    while a >= c:
        if counter:
            a += a / 100 * 15
        a -= b
        counter += 1
    print(counter)


def three():
    segments_coords = []
    for i in range(4):
        a = int(input())
        b = int(input())
        segments_coords.append((a, b))

    n = int(input())
    points = [int(input()) for i in range(n)]

    segments_count = [0, 0, 0, 0]

    for point in points:
        for i, (start, end) in enumerate(segments_coords):
            if start <= point <= end:   
                segments_count[i] += 1
    
    print(segments_count)


def four():
    n = int(input())
    factories = [tuple(map(int, input().split())) for i in range(n)]

    total_power = 0
    for factory in factories:
        total_power += factory[1]

    powerful_factory_number = -1
    for factory in factories:
        if factory[0] > powerful_factory_number:
            powerful_factory_number = factory[0]
    
    print(total_power, powerful_factory_number)


#one()
#two()
three()
four()

File: lab/3-4/lab_3-4.pas
procedure one();
begin
  var n := readinteger();

  writeln('Число' + #9 + 'Квадрат');
  for var i := 1 to n do
  begin
    if i mod 2 <> 0 then
      writeln(i + #9 + sqr(i));
  end;
end;


procedure two();
begin
  var (a, b, c) := readreal3();
  var counter := 0;
  
  while a >= c do
  begin
    if counter <> 0 then
      a += a / 100 * 15;
    a -= b;
    counter += 1;
  end;
  
  writeln(counter);
end;


procedure three();
begin
  var segments_coords: array of (integer, integer);
  var a, b, n, point: integer;
  var points: array of integer;
  var segments_count: array[0..3] of integer;
  
  setlength(segments_coords, 4);
  for var i := 0 to 3 do
  begin
    readln(a);
    readln(b);
    segments_coords[i] := (a, b);
  end;
  
  readln(n);
  setlength(points, n);
  for var i := 0 to n-1 do
    readln(points[i]);
  
  for var i := 0 to 3 do
    segments_count[i] := 0;
  
  for var i := 0 to n-1 do
  begin
    point := points[i];
    for var j := 0 to 3 do
    begin
      if (segments_coords[j].item1 <= point) and (point <= segments_coords[j].item2) then
        segments_count[j] := segments_count[j] + 1;
    end;
  end;
  
  for var i := 0 to 3 do
  begin
    write(segments_count[i], ', ');
  end;
end;

begin
  //one();
  //two();
  three();
end.